shpkpr/marathon/application.py

```diff
@@ -0,0 +1,74 @@
+# stdlib imports
+import json
+import os
+
+# third-party imports
+import jsonschema
+from cached_property import cached_property
+
+# local imports
+from shpkpr import exceptions
+
+
+def default_schema():
+    """Path to the default schema used to validate application definitions
+    before sending to marathon.
+    """
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)), "schema", "app.json")
+
+
+class ValidationError(exceptions.ShpkprException):
+    exit_code = 2
+
+    def format_message(self):
+        return 'Unable to validate application: %s' % self.message
+
+
+class ApplicationError(exceptions.ShpkprException):
+    exit_code = 2
+
+
+class MarathonApplication(object):
```

Comment:
This feels like a bit of a strange object, it doesn't so much model a Marathon Application as act as a hash with some helper functions on it. The code that is using it is accessing it as a deeply nested hash, so it's not really giving any guarantees about the object other than some JSON schema validation.

I think we should either treat the Marathon Application as a hash and have an object that handles the validation of that JSON or model out the Marathon Application as full objects with properties and nested objects.